combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/BleDevice.kt

```diff
@@ -152,6 +163,7 @@ internal open class BleDevice (
                     }
 
                     isConnected.set(connectionState == DeviceConnectionState.CONNECTED)
+                    isDisconnected.set(state is State.Disconnected)
```

Comment:
I don't see any usages where `isConnected` and `isDisconnected` checks could get interleaved, but if there really is some `Atomic`-ness necessary here, I'd make sure that there's not something that could sneak in between these two and end up with both being true.